plugins/meta/libkv/libkv.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2016 Roman Mohr <rmohr@redhat.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is a "meta-plugin". According to its config, it loads a JSON array
+// of types.NetConf from the specified key/value store. Then it delegates
+// one loaded NetConf after the other to the specified plugin.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/containernetworking/cni/pkg/invoke"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+
+	"github.com/docker/libkv"
+	"github.com/docker/libkv/store"
+	"github.com/docker/libkv/store/consul"
+	"github.com/docker/libkv/store/etcd"
+	"github.com/docker/libkv/store/zookeeper"
+	"log"
+	"strings"
+	"time"
+)
+
+const (
+	stateDir = "/var/lib/cni/libkv"
```

Comment:
Under the assumption that containers will go down if the host goes down, and that the CNI config is coupled to a container, then I think that `run` is a better fit. Because: `run` will be emptied on shutdown (like the container is "cleaned" on shutdown).
In the end it really is just about the semantics of `lib` (which survives reboots) vs `run` (which is bound to the runtime of the host). Functional wise there should not be a difference at all.

Said in other words.
You only will be calling DEL if the container is getting deleted in a controlled fashion, and only when it's getting deleted controlled, then you need the CNI config.
If the host is going down unexpected, then - because the container is getting destroyed - you also want to get rid of the related CNI config.

Or am I missing something?